src/main/java/io/debezium/connector/yugabytedb/connection/YugabyteDBConnection.java

```diff
@@ -114,6 +134,19 @@ public YugabyteDBConnection(YugabyteDBConnectorConfig config,
         }
     }
 
+    public YugabyteDBConnection(YugabyteDBConnectorConfig config,
+                                YugabyteDBTypeRegistry yugabyteDBTypeRegistry, String connectionUsage) {
+        this(config,yugabyteDBTypeRegistry, connectionUsage, config.getJdbcConfig().getHostname().contains(":")
```

Comment:
For the code snippet which uses a conditional operator here to select which URL pattern to use, we can create some getter in `YugabyteDBConnectorConfig` itself, for example:
```java
public <connectionFactoryType> getConnectionFactory() {
return getHostname().contains(":")
                        ? JdbcConnection.patternBasedFactory(MULTI_HOST_URL_PATTERN,
                            org.postgresql.Driver.class.getName(),
                            YugabyteDBConnection.class.getClassLoader(),
                            JdbcConfiguration.PORT.withDefault(PORT.defaultValueAsString()))
                        : JdbcConnection.patternBasedFactory(SINGLE_HOST_URL_PATTERN,
                                org.postgresql.Driver.class.getName(),
                                YugabyteDBConnection.class.getClassLoader(), JdbcConfiguration.PORT
                                        .withDefault(PORT.defaultValueAsString()))
}
```

And then replace all the instances of this conditional operator with `config.getConnectionFactory()`